src/client/Microsoft.Identity.Client/Utils/DateTimeHelpers.cs

```diff
@@ -63,15 +63,39 @@ public static long GetDurationFromWindowsTimestamp(string windowsTimestampInFutu
             return (long)unixTimestamp - CurrDateTimeInUnixTimestamp();
         }
 
-        public static long GetDurationFromNowInSeconds(string unixTimestampInFuture)
+        public static long GetDurationFromNowInSeconds(string dateTimeStamp)
```

Comment:
I am not sure this is the right place for this fix. `GetDurationFromNowInSeconds(string unixTimestampInFuture)` works fine for ESTS and broker responses. I suggest you create a separate helper for MSI and use that in https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/main/src/client/Microsoft.Identity.Client/OAuth2/MsalTokenResponse.cs#L255

Note that this is all in the hot path.